gulp-tasks/puppeteer/build_docs.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * @fileoverview Generates the Puppeteer docs at /web/tools/puppeteer/
+ *
+ * @author Eric Bidelman <e.bidelman@google.com>
+ */
+
+'use strict';
+
+const gulp = require('gulp');
+const os = require('os');
+const path = require('path');
+const fs = require('fs-extra');
+const jsYaml = require('js-yaml');
+const {promisify} = require('util');
+const semver = require('semver');
+
+const checkoutRepoAtTag = require('../reference-docs/get-source-code');
+const wfHelper = require('../wfHelper');
+
+const readFile = promisify(fs.readFile);
+const ensureDir = promisify(fs.ensureDir);
+const writeFile = promisify(fs.writeFile);
+
+const ORG = 'GoogleChrome';
+const USER = 'puppeteer';
+const GITHUB_PROJECT_URL = `https://github.com/${ORG}/${USER}`;
+const GIT_URL = `${GITHUB_PROJECT_URL}.git`;
+
+const DOC_FILES = [
+  'README.md',
+  // 'docs/api.md',
+  'docs/troubleshooting.md',
+  'examples/README.md',
+];
+
+const PPTR_FOLDER_PATH = path.join(
+  __dirname, '..', '..', 'src', 'content', 'en', 'tools', 'puppeteer'
+);
+
+/**
+ * Writes a file to disk.
+ * @param {string} outputFile Path to output file
+ * @param {string} content File content to write.
+ */
+async function saveFile(outputFile, content) {
+  const filePath = path.join(PPTR_FOLDER_PATH, '_src', outputFile);
+  await ensureDir(path.dirname(filePath));
+  await writeFile(filePath, content);
+}
+
+/**
+ * Updates _toc.yaml file with latest releases and saves file to disk.
+ * @param {string} tocYaml Path to toc yaml file.
+ * @param {!Array<string>} tags
+ */
+async function generateTOC(tocYaml, tags) {
+  const yamlContent = await readFile(tocYaml, {encoding: 'utf-8'});
+  const yamlJSON = jsYaml.safeLoad(yamlContent);
+
+  // Update/add API Reference section to TOC.
+  let apiIdx = yamlJSON.toc.findIndex((item) => item.title === 'API Reference');
+  if (apiIdx === -1) {
+    apiIdx = yamlJSON.toc.findIndex((item) => item.title === 'Overview') + 1;
+    yamlJSON.toc.splice(apiIdx, 0, {
+      title: 'API Reference',
+      style: 'accordion',
+      section: [],
+    });
+  }
+
+  yamlJSON.toc[apiIdx].section = tags.map((tag, i) => {
+    const title = i === 0 ? `${tag} (latest}` : tag;
+    return {
+      title,
+      section: [{
+        title: 'Docs',
+        path: `${GITHUB_PROJECT_URL}/blob/${tag}/docs/api.md`,
+        status: 'external',
+      }, {
+        title: 'Release Notes',
+        path: `${GITHUB_PROJECT_URL}/releases/tag/${tag}`,
+        status: 'external',
+      }],
+    };
+  });
+
+  await writeFile(tocYaml, jsYaml.safeDump(yamlJSON));
```

Comment:
This feels error prone. If anything subtle goes wrong or changes, there is a chance it'll work locally and not on devsite. using includes like Pete mentioned would be a much better approach.

Reply:
done